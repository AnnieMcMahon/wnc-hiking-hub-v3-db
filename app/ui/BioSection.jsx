export default function BioSection({user, onClick}) {
  return (
    <div className="bio-section">
        <div className="bio-header-section">
          <img className="avatar" src={user.avatar} alt="avatar"/>
          <h1>{user.user_name}</h1>
          <button onClick={onClick}>Edit Bio</button>
        </div>
        <div className="bio-text-section">
          <h2>About Me</h2>
          <div id="bio-text" className="text-box">
            <p>{user.bio}</p>
          </div>
        </div>
      </div>
  );
};