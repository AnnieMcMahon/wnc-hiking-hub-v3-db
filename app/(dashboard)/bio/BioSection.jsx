import { DEFAULT_USER_NAME, DEFAULT_BIO, DEFAULT_AVATAR } from "@/app/lib/constants";
import { Button } from "@/components/ui/button";

export default function BioSection({ user = {}, onClick = () => {} }) {
  const avatar = user.avatar || DEFAULT_AVATAR;
  const userName = user.user_name || DEFAULT_USER_NAME;
  const bio = user.bio || DEFAULT_BIO;
  const handleClick = () => {onClick()};
  return (
    <div className="bio-section">
        <div className="bio-header-section">
          <img className="avatar" src={avatar} alt="avatar"/>
          <h1>{userName}</h1>
          <Button onClick={handleClick}>Edit Bio</Button>
        </div>
        <div className="bio-text-section">
          <h2>About Me</h2>
          <div id="bio-text" className="text-box">
            <p className="text-sm text-left pt-2">{bio}</p>
          </div>
        </div>
      </div>
  );
};