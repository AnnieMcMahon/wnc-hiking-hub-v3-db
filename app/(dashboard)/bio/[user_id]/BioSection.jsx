"use client";

import { DEFAULT_USER_NAME, DEFAULT_BIO, DEFAULT_AVATAR } from "@/app/lib/constants";
import { Button } from "@/components/ui/button";
import { useGlobal } from "@/app/context/GlobalContext";

export default function BioSection({ user = {}, onClick = () => {} }) {
  const { currentUser } = useGlobal();
  const avatar = user.avatar || DEFAULT_AVATAR;
  const userName = user.user_name || DEFAULT_USER_NAME;
  const bio = user.bio || DEFAULT_BIO;
  const showEditButton = user.id == currentUser.id;
  const handleClick = () => {onClick()};
  return (
    <div className="w-[95%] mx-2 md:w-[60%]">
        <div className={`flex items-center gap-4 m-2 md:m-4 ${showEditButton ? "justify-between" : "justify-left"}`}>
          <img className="h-8 md:h-16 w-8 md:w-16 border border-green-800 rounded-full" src={avatar} alt="avatar"/>
          <h1>{userName}</h1>
          {showEditButton &&
          <Button onClick={handleClick}>Edit Bio</Button>
        }
        </div>
        <div className="px-2 md:px-16">
          <h2>About Me</h2>
          <div className="text-box min-h-40">
            <p className="text-sm text-left pt-2">{bio}</p>
          </div>
        </div>
      </div>
  );
};