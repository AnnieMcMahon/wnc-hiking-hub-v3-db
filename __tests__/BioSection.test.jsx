import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import BioSection from "@/app/ui/BioSection";

const user = {
  email: "user@gmail.com",
  avatar: "/newUser.png",
  user_name: "New User",
  bio: "Tell us about yourself"
};

describe("BioSection", () => {
  it("renders a BioSection component", () => {
    render(<BioSection user={user}/>);
  });

  it("renders a title", () => {
    render(<BioSection user={user} />);
    screen.getByText("About Me");
    screen.debug;
  });
});