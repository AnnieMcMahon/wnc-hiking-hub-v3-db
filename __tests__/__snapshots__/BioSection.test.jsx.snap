// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioSection matches the snapshot 1`] = `
<div>
  <div
    class="bio-section"
  >
    <div
      class="bio-header-section"
    >
      <img
        alt="avatar"
        class="avatar"
        src="/newUser.png"
      />
      <h1>
        Demo User
      </h1>
      <button>
        Edit Bio
      </button>
    </div>
    <div
      class="bio-text-section"
    >
      <h2>
        About Me
      </h2>
      <div
        class="text-box"
        id="bio-text"
      >
        <p>
          Bio text goes here ... Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem reprehenderit assumenda quibusdam iste eos exercitationem molestias nulla fuga repellat incidunt ullam cupiditate necessitatibus quidem debitis dolorum nesciunt labore quos, laboriosam tempora qui tempore quo velit! Libero quod eligendi at eum.
        </p>
      </div>
    </div>
  </div>
</div>
`;
